### PR TITLE
Add AWS::EC2::CustomerGateway support

### PR DESCRIPTION
**Pull request by adamfokken**

### CODE REVIEW

File: src/main/scala/com/monsanto/arch/cloudformation/model/resource/EC2.scala [hunk 1]
```diff
@@ -78,6 +78,20 @@ object `AWS::EC2::KeyPair::KeyName` extends DefaultJsonProtocol {
   implicit val format: JsonFormat[`AWS::EC2::KeyPair::KeyName`] = jsonFormat2(`AWS::EC2::KeyPair::KeyName`.apply)
 }
 
+case class `AWS::EC2::CustomerGateway`(
+  name: String,
+  BgpAsn: Int,
+  IpAddress: String,
```
Comment (tj-corrigan) [inline src/main/scala/com/monsanto/arch/cloudformation/model/resource/EC2.scala]: Can you make this more typesafe by using the new `IPAddress` type ( see https://github.com/MonsantoCo/cloudformation-template-generator/blob/master/src/main/scala/com/monsanto/arch/cloudformation/model/resource/EC2.scala#L235-L248).

**Comment by adamfokken** [comment]
I think I have it using IPAddress now.

**Comment by tj-corrigan** [comment]
Please squash to a single commit and I'll get it merged and released.

**Comment by adamfokken** [comment]
The commits should be squashed.

**Pull request merged**